### PR TITLE
[Feature] Download multiple files as ZIP

### LINKED ISSUE #11533: ✨ When downloading multiple applicants or users from a table, bundle them as individual files zipped together
## ✨ Feature

When selecting multiple candidate or user profiles for download, they're currently pasted into a single long word document.

We would prefer to generate a separate file for each candidate/user, and then bundle them into a zip file with actually downloaded by the user.

## ✅ Acceptance Criteria

- [ ] If the pool candidate or user profile download includes multiple users, each user/candidate gets a separate file, and they are bundled into a single zip file for download.
- [ ] If only a single user/candidate would be downloaded, it gets downloaded as a single unzipped file.

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #11533 

## 👋 Introduction

Updates file generation to use ZIP when downloading multiple entities.

## 🕵️ Details

The scope of this expanded a bit once we realized we needed to add a unique identifier to the file names and some extra requests were made:

- Remove hyphen between first and last name
- Add incrementing number to duplicate file names
- Remove need to pass locale

## 🧪 Testing

1. Build app `pnpm run dev:fresh`
2. Start queue `make queue-work`
3. Login as admin `admin@test.com`
4. Download a version of each file
5. Confirm you get a Zip for multiple items
6. Confirm all other downloads still function as expected

### CODE REVIEW

**Comment by petertgiles** [comment]
The code looks great!  I love getting the locale from the app now.  I'm having some real trouble running it though.
I'm getting some errors when I navigate to the _Talent placement_ tab of a process.  I think this is preexisting, though, and logged a bug. [11601](https://github.com/GCTC-NTGC/gc-digital-talent/issues/11601)

The second problem was when I tried to download a single file, which is maybe from this PR?
```
[2024-09-24 14:36:41] local.INFO: Request from platform admin, admin@test.com, http://localhost:8000/en/admin/pools/07b436b6-8e82-4489-b150-04cf16d8dcba/pool-candidates, {"operationName":"DownloadPoolCandidateDoc","query":"mutation DownloadPoolCandidateDoc($id: UUID!, $anonymous: Boolean!) {\n  downloadPoolCandidateDoc(id: $id, anonymous: $anonymous)\n}","variables":{"anonymous":false,"id":"51045810-4980-42f8-b4a7-5dc0a743cdfe"}}  
[2024-09-24 14:36:41] local.ERROR: Error starting candidate document generation Attempted to lazy load [skill] on model [App\Models\UserSkill] but lazy loading is disabled. 
```

**Review by petertgiles — CHANGES_REQUESTED** [review]
Small code things.

File: api/app/Traits/Generator/GeneratesUserDoc.php [hunk 1]
```diff
@@ -270,10 +270,10 @@ protected function experiences(Section $section, Collection $experienceCollectio
      * @param  Section  $section  The section to add info to
      * @param  Experience  $experience  The experience being generated
      * @param  string  $type  The type of experience being generated
-     * @param  int  $headingRank  The rank of the heading
```
[the remark this answers is not in the document]
Comment (esizer) [reply api/app/Traits/Generator/GeneratesUserDoc.php:273]: This was actually pint that got rid of these. I'm not sure why, but it seems like they don't use PHPDoc for items that have a fallback? :thinking:

**Comment by esizer** [comment]
> The code looks great! I love getting the locale from the app now. I'm having some real trouble running it though.
> I'm getting some errors when I navigate to the _Talent placement_ tab of a process. I think this is preexisting, though, and logged a bug. [11601](https://github.com/GCTC-NTGC/gc-digital-talent/issues/11601)

Not sure why `load` is working but `loadMissing` was not since it seems it was clear they were missing. But this should be fixed now after [eager load skills](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11583/commits/c1ac324255b79804470f85aee9497e40fccd0465)

> The second problem was when I tried to download a single file, which is maybe from this PR?

Huh. weird that was removed... Added it back in [add missing import](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11583/commits/4be2420f16a53ac3df478b69a8e31fb09088b8a8)

**Review by petertgiles — APPROVED** [review]
Nice!  It's working great.  I tried it on DEV, too.

**Pull request merged**